File: app/src/main/java/crux/bphc/cms/helper/FileUtils.java
package crux.bphc.cms.helper;

import android.webkit.MimeTypeMap;

import androidx.annotation.NonNull;

import crux.bphc.cms.R;

public class FileUtils {

    private FileUtils() {}

    public static int getDrawableIconFromFileName(String filename) {
        String mimeType = getFileMimeType(filename);
        if (mimeType == null)
            return -1;

        switch (mimeType) {
            case "application/pdf":
                return (R.drawable.file_pdf);

            case "application/vnd.ms-excel":
            case "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet":
                return (R.drawable.file_excel);

            case "application/msword":
            case "application/vnd.openxmlformats-officedocument.wordprocessingml.document":
                return (R.drawable.file_word);

            case "application/vnd.ms-powerpoint":
            case "application/vnd.openxmlformats-officedocument.presentationml.presentation":
                return (R.drawable.file_powerpoint);

            default:
                return -1;
        }
    }

    public static String getFileMimeType(String filename) {
        return MimeTypeMap.getSingleton().getMimeTypeFromExtension(getExtension(filename));
    }

    @NonNull
    public static String getExtension(String filename) {
        return filename.substring(filename.lastIndexOf('.') + 1);
    }
}

File: app/src/main/java/crux/bphc/cms/utils/FileUtils.kt
package crux.bphc.cms.utils

import android.webkit.MimeTypeMap
import crux.bphc.cms.R

object FileUtils {
    fun getDrawableIconFromFileName(filename: String): Int {
        val mimeType = getFileMimeType(filename) ?: return R.drawable.ic_document_unknown_type
        return when (mimeType) {
            "application/pdf" -> R.drawable.file_pdf
            "application/vnd.ms-excel", "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet" -> R.drawable.file_excel
            "application/msword", "application/vnd.openxmlformats-officedocument.wordprocessingml.document" -> R.drawable.file_word
            "application/vnd.ms-powerpoint", "application/vnd.openxmlformats-officedocument.presentationml.template", "application/vnd.openxmlformats-officedocument.presentationml.slideshow", "application/vnd.oasis.opendocument.presentation" -> R.drawable.file_powerpoint
            else -> R.drawable.ic_document_unknown_type
        }
    }

    fun getFileMimeType(filename: String): String? {
        return MimeTypeMap.getSingleton().getMimeTypeFromExtension(getExtension(filename))
    }

    private fun getExtension(filename: String): String {
        return filename.substring(filename.lastIndexOf('.') + 1)
    }
}